### ISSUE TITLE
FTBFS with boost-1.60.0

### ISSUE THREAD

**Issue by bavier** [csound/csound#570]
Building Csound with boost-1.60.0 produces the following compilation error

```
In file included from /tmp/nix-build-csound-6.05.drv-1/Csound6.05/./include/csdl.h:110:0,
                 from /tmp/nix-build-csound-6.05.drv-1/Csound6.05/./include/OpcodeBase.hpp:6,
                 from /tmp/nix-build-csound-6.05.drv-1/Csound6.05/Opcodes/chua/ChuaOscillator.cpp:120:
/tmp/nix-build-csound-6.05.drv-1/Csound6.05/./include/interlocks.h:40:12: error: expected identifier before ‘(’ token
 #define CB (0x0060)
            ^
/gnu/store/qr7v564lw46j7ak6k5va7l0kll77hfs5-boost-1.60.0/include/boost/type_traits/detail/composite_member_pointer_type.hpp:89:26: note: in expansion of macro ‘CB’
 template<class CT, class CB>
                          ^
/tmp/nix-build-csound-6.05.drv-1/Csound6.05/./include/interlocks.h:40:13: error: expected ‘)’ before numeric constant
 #define CB (0x0060)
             ^
```

The issue seems to be that in `Opcodes/chua/ChuaOscillator.cpp` the `include/csdl.h` gets included before the boost headers, and the definition of the macro `CB` in the former interferes with template declarations in latter.

A simple fix might be to change the order of the includes:

```
--- Csound6.05/Opcodes/chua/ChuaOscillator.cpp	2015-04-25 14:06:22.995646234 -0500
+++ Csound6.05/Opcodes/chua/ChuaOscillator.cpp	2015-12-21 10:31:58.182362640 -0600
@@ -117,11 +117,12 @@
 //     d = sys_variables(12);
 //     gnor = a*(x.^3) + b*(x.^2) + c*x + d;

-#include <OpcodeBase.hpp>
 #include <boost/numeric/ublas/vector.hpp>
 using namespace boost::numeric;
 #include <cmath>

+#include <OpcodeBase.hpp>
+
 #undef CS_KSMPS
 #define CS_KSMPS     (opds.insdshead->ksmps)

```

**Comment by jpffitch** [comment]
Believed fixed in git, but I do not have tt recent a boost; please check

**Comment by ainola** [comment]
I can confirm the fix (boost 1.60.0, archlinux)
